src/components/PopupWithForm.js

```diff
@@ -22,13 +23,17 @@ export default class PopupWithForm extends Popup {
   setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (evt) => {
-      evt.preventDefault();
+      // evt.preventDefault();
       const inputValues = this._getInputValues();
       this._handleFormSubmit(inputValues);
-      this.close();
     });
   }
 
+  //
+  setSubmitAction(action) {
```

Comment:
Needs correction: the PopupWithForm class should not have such method, it's the logic of confirmation popup

Reply:
Got it.